tools/sof-tplgreader.py

```diff
@@ -95,7 +98,14 @@ def loadFile(self, filename, sofcard=0):
         # format pipeline, this change for script direct access 'rate' 'channel' 'dev' 'snd'
         for pipeline in self._pipeline_lst:
             #pipeline['fmt']=pipeline['fmt'].upper().replace('LE', '_LE')
-            pipeline['rate'] = pipeline['rate_min'] if int(pipeline['rate_min']) != 0 else pipeline['rate_max']
+            # set rate in order of rate_min, rate_max. If both are 0, get first
+            # sampling rate(rates[0]) from supported rates
+            if int(pipeline['rate_min']) != 0:
+                pipeline['rate'] = pipeline['rate_min']
+            elif int(pipeline['rate_max']) != 0:
+                pipeline['rate'] = pipeline['rate_max']
+            elif rates[0] != 0:
+                pipeline['rate'] = pipeline['rates'].split(' ')[0]
```

Comment:
I think the order should be reversed. check if rates is not empty and only then check min/max

Reply:
And also what if there's no rates or rate_min/rate_max in tplg? Can we catch that and throw an error is rate is 0?